cardano-cli/src/Cardano/CLI/Conway/Commands.hs

```diff
@@ -35,16 +43,40 @@ runGovernanceCreateVoteCmd
   -> VoteChoice
   -> VoterType
   -> TxIn
-  -> StakeIdentifier
+  -> VerificationKeyOrFile StakePoolKey
   -> ConwayVoteFile Out
   -> ExceptT GovernanceCmdError IO ()
-runGovernanceCreateVoteCmd (AnyShelleyBasedEra sbe) vChoice cType govActionTxIn votingStakeCred oFp = do
-  stakeCred <- firstExceptT StakeCredGovCmdError $ getStakeCredentialFromIdentifier votingStakeCred
-  votingCred <- hoistEither $ first VotingCredentialDecodeGovCmdEror $ toVotingCredential sbe stakeCred
+runGovernanceCreateVoteCmd (AnyShelleyBasedEra sbe) vChoice cType govActionTxIn stakePoolVerKeyOrFile oFp = do
+  stakePoolVerKey <- firstExceptT VotingCredentialDecodeError
+    . newExceptT
+    $ readVerificationKeyOrFile AsStakePoolKey stakePoolVerKeyOrFile
+
+  let stakePoolVerificationKeyHash = unStakePoolKeyHash $ verificationKeyHash stakePoolVerKey
+  votingCred <- hoistEither $ first VotingCredentialDecodeGovCmdEror $ eraDecodeVotingCredential sbe $ Plain.serialize $ Ledger.KeyHashObj stakePoolVerificationKeyHash
+
   let govActIdentifier = makeGoveranceActionIdentifier sbe govActionTxIn
       voteProcedure = createVotingProcedure sbe vChoice cType govActIdentifier votingCred
   firstExceptT WriteFileError . newExceptT $ obtainEraPParamsConstraint sbe $ writeFileTextEnvelope oFp Nothing voteProcedure
+  where
```

Comment:
Both functions in `where` to be removed - placed here temporarily to contain the changes in `cardano-cli` only for now.